### ISSUE TITLE
sf.IndexHierarchy.from_labels silently discards incorrectly sized labels

### ISSUE THREAD

**Issue by ForeverWintr** [InvestmentSystems/static-frame#212]
When creating an IndexHierarchy using `from_labels`, incorrectly sized labels are dropped:

```python
labels = ((1, 2), (3,), (4, 5))
sf.IndexHierarchy.from_labels(labels)
<IndexHierarchy>
1                2.0
4                5.0
<int64>          <float64>
```

Note that the middle label has disappeared (and the lower level has been coerced to float, maybe because of a `nan`).

I think this behavior is undesirable, although I'm not sure what should happen instead. An explicit failure is probably safest, but in the case in which I encountered this, filling the 'short' label to all missing levels is desirable, e.g. producing:

```python
sf.IndexHierarchy.from_labels(labels)
<IndexHierarchy>
1                2
3                3
4                5
<int64>          <int64>
```

**Comment by flexatone** [comment]
Many thanks for isolating this issue. I think this should raise an exception. I will investigate further.

**Comment by flexatone** [comment]
This is fixed and will be included in the 0.6.18 release. Thanks again for isolating this issue.

```python
>>> labels = ((1, 2), (3,), (4, 5))
>>> sf.IndexHierarchy.from_labels(labels)
Traceback (most recent call last):
static_frame.core.exception.ErrorInitIndex: Inconsistent label depth: expected 2, got 1
```
